.github/workflows/codeql-analysis.yml

```diff
@@ -14,8 +14,15 @@ on:
   schedule:
     - cron: '0 13 * * 3'
 
+permissions:
+  contents: read
+
```

Comment:
The default permissions for our workflows is already 'read', so I don't believe this is necessary.

Reply:
@di this also sets all the other (non-`content`) scopes to `null` restricting them even further. It's one of the latest best practices security-wise.

Reply:
[It looks like](https://github.com/search?q=org%3Apypa+path%3A.github+permissions) this isn't widely-adopted in the pypa org; only pipenv seems to do it for their CI workflow. @neilnaveen out of curiosity, why did you choose this to add this to Twine?

Reply:
@bhrutledge I saw it being adopted by attrs and I know that Tidelift is starting to push for openSSF, so I foresee it appearing on our radars more often over time. I'm slowly adopting this too.